course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -128,19 +160,540 @@ describe('Unit testing PeopleController', function() {
             var data = 'fasfeed';
             var result = scope.renderSource(data, undefined, undefined, undefined);
-            expect(result).toBe('Registrar Added');
+            expect(result).toEqual('Registrar Added');
         });
 
         it('renderSource for manual', function() {
             var data = '';
             var result = scope.renderSource(data, undefined, undefined, undefined);
-            expect(result).toBe('Manually Added');
+            expect(result).toEqual('Manually Added');
+        });
+    });
+
+    describe('controller methods which act purely locally', function() {
+        // group these so that we can init the scope/controllers/directives
+        // such that we're just testing the controller methods
+        beforeEach(function() {
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'methods without side effects tests',
+            };
+            courseInstances.instances[ci.course_instance_id] = ci;
+            controller = $controller('PeopleController', {$scope: scope});
+        });
+
+        describe('clearSearchResults', function() {
+            it('should empty any search results', function() {
+                scope.searchResults = [{}];  // contents don't matter
+                scope.clearSearchResults();
+                expect(scope.searchResults).toEqual([]);
+            });
+        });
+
+        describe('closeAlert', function() {
+            it('should work when an alert is present', function() {
+                scope.testAlerts = [{}];  // contents don't matter
+                scope.closeAlert('testAlerts', 0);
+                expect(scope.testAlerts).toEqual([]);
+            });
+            
+            it('should not blow up when no alert is present', function() {
+                scope.testAlerts = [];
+                scope.closeAlert('testAlerts', 0);
+                expect(scope.testAlerts).toEqual([]);
+            });
+
+            it('should only remove the index requested', function() {
+                scope.testAlerts = [1, 2, 3];
+                scope.closeAlert('testAlerts', 1);
+                expect(scope.testAlerts).toEqual([1,3]);
+            });
+        });
+
+        describe('compareRoles', function() {
+            it('should sort as expected', function() {
+                var roles = [
+                    {active: 0, prime_role_indicator: ''},
+                    {active: 1, prime_role_indicator: ''},
+                    {active: 0, prime_role_indicator: 'N'},
+                    {active: 1, prime_role_indicator: 'N'},
+                    {active: 0, prime_role_indicator: 'Y'},
+                    {active: 1, prime_role_indicator: 'Y'},
+                ];
+                var afterSorting = [
+                    {active: 1, prime_role_indicator: 'Y'},
+                    {active: 1, prime_role_indicator: 'N'},
+                    {active: 1, prime_role_indicator: ''},
```

Comment:
Looking at this now, I wonder if perhaps `''` should be ranked higher than `'N'`... maybe a `''` is a sort of implicit `'Y'` while `'N'` is more explicitly not `'Y'`... anyways. Probably there's not a soul at Harvard that can address that with confidence. :)